### DISCUSSION TITLE
Streaming signature for display on another screen, and signature playback.

### DISCUSSION THREAD

**Discussion by paulferrett** [szimek/signature_pad#682] [Ideas]
Hi there! Great library, thanks for your work :) 

I'm looking to do two things with the library and I'm not sure how:
1. Stream the curve as its being draw down a websocket connection so it can be displayed on another browser window - I'm looking to use the main screen as the UI, and a mobile device as just a signing pad; and also
2. Capture the data in 1 to use as an animated signature for signing replay.

It seems the event listeners trigger the raw events, but don't give access to the curve that's generated? What would you suggest I try? I'd be happy to code up some changes and submit a PR if you think this would be useful?

Thank you!

**Comment by UziTech** [comment]
If you want access to the points you can get them from `signaturePad.toData()`.

You can see an example at https://jsfiddle.net/UziTech/soux13av/8/

> It seems the event listeners trigger the raw events, but don't give access to the curve that's generated

If we want to return other data in the event detail I would be ok with that. It looks like we should have the point that is created sent with `afterUpdateStroke` and the last point group sent with `endStroke`.

**Comment by paulferrett** [comment]
Oh that's just perfect - thank you so much. I think I would just grab the newer events and record those / send those down a websocket and no need to update the event. Thanks again for your help and the great library. 🙏🏻
